test/unit/three_scale/invitation_email_delivery_observer_test.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ThreeScale::InvitationEmailDeliveryObserverTest < ActiveSupport::TestCase
+  include ActionMailer::TestHelper
+
+  def setup
+    InvitationMailer.register_observer ThreeScale::InvitationEmailDeliveryObserver
+  end
+
+  def test_update_date_after_sent_only
+    @mailer = nil
+
+    assert_emails 0 do
+      invitation = FactoryBot.create(:invitation)
+      invitation.expects(:update_column).once
+
+      Invitation.expects(find_by).once.return(invitation)
```

Comment:
imo, it is better without mocking this... it does not matter how it does it internally (calling to which methods), what matters is the final result, and this is what it should test... so imo you should use `Timecop` and then test the final date of the attribute
Example:
https://github.com/3scale/porta/blob/ad6a20c26bd4e53e68b46a4a9edafac6d801f87f/test/events/zync_event_test.rb#L64-L70

```ruby
Timecop.freeze(date) do
  ### here whatever u want to execute being the current date/time "date"
end
```